galasa-ui/docs/date-time-handling.md

```diff
@@ -0,0 +1,63 @@
+# Guide to Date and Timezone Handling in the Galasa UI
+This document outlines the standardized approach for handling all date and time-related operations within the codebase. The goal is to eliminate timezone ambiguity, ensure data consistency between the UI and server, and provide a clear and intuitive experience for users, regardless of their location.
+
+The **core principle** of our date handling strategy is simple:
+> Store and process all dates in UTC. Display all dates to the user via the `formatDate` utility.
+
+Adhering to this rule prevents a wide range of common bugs related to timezones and Daylight Saving Time.
+
+## What's New?
+- A new timezone selection section is available in the user settings page.
+- All dates and time displays throughout the UI have been updated to respect the user's selected timezone preference (or their browser's default)
+- Date formatting logic has been consolidated into a central React context to ensure consistency
+
+## Core Concepts & Implementation
+The entire system is built around the `DateTimeFormatContext`.
+- **`DateTimeFormatProvider`**: This component must wrap the entire application (e.g., in `layout.tsx`). It is responsible for initializing user preferences from `localStorage` and making them available to all child components.
+- **`useDateTimeFormat`**: This custom hook is the primary way components should interact with the system. It provides access to the user's preferences and, most importantly, the formatting utilities.
+
+> [! Note] 
+> Preferences are currently saved in the localStorage but we plan to move all preferences to be saved in the server per user. 
+> 
+
+### How it Works
+1.  **Internal Standard: UTC**: All dates received from the API are expected to be in UTC (typically as an ISO 8601 string, e.g., `"2023-10-26T12:00:00.000Z"`). When we create new `Date` objects in the code, they represent a universal point in time, independent of any timezone.
+2. **Displaying Dates to the User**: To show a date in the UI, you must use the `formatDate` function from the `useDateTimeFormat` hook.
+```typescript
+const { formatDate } = useDateTimeFormat(); 
+```
+
+The formatDate function automatically handles everything:
+- It reads the user's chosen timezone (e.g., 'America/New_York') from the context.
+- It reads the user's chosen locale (e.g., 'en-GB') for correct date ordering (DD/MM/YYYY).
+- It reads the user's time format preference (12-hour or 24-hour).
+- It combines these to produce a correctly formatted, human-readable string.
+
+## UI-to-UTC Conversion Utilities
+In some cases, especially when a user is **inputting** a specific time, we need to perform the conversion in reverse: from UI-friendly parts into a single UTC Date object. Two utility functions exist for this purpose.
+
+#### `extractDateTimeForUI(date, timezone)`
```

Comment:
These two functions might be better if they are called something like `dateTimeUTC2Local` and `dateTimeLocal2UTC` ?

Reply:
just a suggestion really.